### ISSUE TITLE
Prometheus output sometimes returns no content when using gnmi-cache

### ISSUE THREAD

**Issue by jhujhiti** [karimra/gnmic#530]
I'm having a very unusual issue with the Prometheus output sometimes returning no content. I'm new to gnmic, but I think the following config should be working. Its goal is to add the interface description as a tag on other interface metrics, which should then result in the Prometheus metrics getting the description as an associated label:

```
debug: true
skip-verify: true
targets:
  lab1:
    address: xxx:50052
    username: xxx
    password: xxx
    timeout: 2s
    skip-verify: true
subscriptions:
  srlinux:
    paths:
      - /interface[name=mgmt0]/description
      - /interface[name=mgmt0]/statistics
    mode: STREAM
    stream-mode: SAMPLE
    encoding: JSON_IETF
    sample-interval: 60s
processors:
  group-by:
    event-group-by:
      tags:
        - target
        - interface_name
      by-name: true
      debug: true
  description:
    event-to-tag:
      value-names:
        - '/srl_nokia-interfaces:interface/description'
outputs:
  prom:
    type: prometheus
    listen: :9804
    path: /metrics
    expiration: 600s
    metric-prefix: gnmic
    append-subscription-name: false
    override-timestamps: false
    export-timestamps: false
    strings-as-labels: false
    gnmi-cache: true
    timeout: 10s
    debug: true
    event-processors:
      - group-by
      - description

```

Sometimes, this configuration works great and I get Prometheus metrics that look exactly like what I expect, for instance: `gnmic_srl_nokia_interfaces_interface_statistics_carrier_transitions{description="hello world",interface_name="mgmt0",subscription_name="srlinux",target="lab1"} 2`. But, sometimes, a GET against /metrics here simply returns nothing at all. No new data has been fetched from the device according to the debug logs, and there are no errors reported in the logs either.

I have also noticed that the issue happens much more frequently, but still not all the time, when `strings-as-labels` is enabled instead of disabled like it is above. Here's a quick comparison:

`strings-as-labels` disabled:
```
% while true
do ; echo tick ;
curl -vo/dev/null http://localhost:9804/metrics 2>&1 | grep 'Content-Length: 0'
sleep 5 
done
tick
tick
tick
tick
tick
tick
tick
tick
tick
< Content-Length: 0
tick
tick
tick
tick
tick
tick
tick
tick
tick
tick
```

And, enabled:
```
% while true
do ; echo tick ;
curl -vo/dev/null http://localhost:9804/metrics 2>&1 | grep 'Content-Length: 0'
sleep 5
done
tick
< Content-Length: 0
tick
< Content-Length: 0
tick
< Content-Length: 0
tick
< Content-Length: 0
tick
tick
< Content-Length: 0
tick
tick
< Content-Length: 0
```

**Comment by karimra** [comment]
Thanks for reporting this, I can reproduce this issue.
I will fix it and release as part of 0.22.1 shortly.

**Comment by jhujhiti** [comment]
Amazing, thank you!

**Comment by jhujhiti** [comment]
Confirmed fixed in 0.22.1

**Comment by karimra** [comment]
Thank you for a well explained issue.
